cpp/dolfinx/fem/DirichletBC.cpp

```diff
@@ -271,6 +273,114 @@ get_remote_bcs2(const common::IndexMap& map0, int bs0,
   return dofs;
 }
 //-----------------------------------------------------------------------------
+/// Find DOFs on this processes that are constrained by a Dirichlet
+/// condition detected by another process
+///
+/// @param[in] map The IndexMap with the dof layout
+/// @param[in] bs The block size of the index map
+/// @param[in] dofs_local The IndexMap with the dof layout
+/// @return List of local dofs with boundary conditions applied but
+///   detected by other processes. It may contain duplicate entries.
+std::vector<std::int32_t>
+get_remote_bcs1_sub(const common::IndexMap& map, int bs,
+                    const std::vector<std::int32_t>& dofs_local)
+{
+  // NOTE: assumes that dofs are unrolled, i.e. not blocked. Could it be
+  // make more efficient to handle the case of a common block size?
+
+  dolfinx::MPI::Comm comm
+      = create_symmetric_comm(map.comm(common::IndexMap::Direction::forward));
+
+  int num_neighbors(-1), outdegree(-2), weighted(-1);
+  MPI_Dist_graph_neighbors_count(comm.comm(), &num_neighbors, &outdegree,
+                                 &weighted);
+  assert(num_neighbors == outdegree);
+
+  // Return early if there are no neighbors
+  if (num_neighbors == 0)
+    return {};
+
+  // Figure out how many entries to receive from each neighbor
+  const int num_dofs = dofs_local.size();
+  std::vector<int> num_dofs_recv(num_neighbors);
+  MPI_Request request;
+  MPI_Ineighbor_allgather(&num_dofs, 1, MPI_INT, num_dofs_recv.data(), 1,
+                          MPI_INT, comm.comm(), &request);
+
+  // NOTE: we consider only dofs that we know are shared
+  // Build array of global indices of dofs
+  std::vector<std::int64_t> dofs_global;
+  dofs_global.reserve(dofs_local.size());
+
+  // Convert dofs indices to 'blocks' relative to index map
+  std::vector<std::int32_t> dofs_local_block = dofs_local;
+  std::for_each(dofs_local_block.begin(), dofs_local_block.end(),
+                [bs](std::int32_t& n) { return n /= bs; });
+
+  // Get global index of each block
+  std::vector<std::int64_t> dofs_global_block(dofs_local_block.size());
+  map.local_to_global(dofs_local_block, dofs_global_block);
+
+  // Convert from block to actual index
+  for (std::size_t j = 0; j < dofs_local.size(); ++j)
+  {
+    const int index_offset = dofs_local[j] % bs;
+    dofs_global.push_back(bs * dofs_global_block[j] + index_offset);
+  }
+
+  // Compute displacements for data to receive. Last entry has total
+  // number of received items.
+  MPI_Wait(&request, MPI_STATUS_IGNORE);
+  std::vector<int> disp(num_neighbors + 1, 0);
+  std::partial_sum(num_dofs_recv.begin(), num_dofs_recv.end(),
+                   std::next(disp.begin()));
+
+  // NOTE: we could use MPI_Neighbor_alltoallv to send only to relevant
```

Comment:
Which line is this comment referring to?

Reply:
Good question, i think this is a remainder from my copy-paste of the similar functions `get_remote_bcs` and `get_remote_bcs2` that has the same comment.